Modules/posixmodule.c

```diff
@@ -8656,7 +8662,12 @@ os_ptsname_impl(PyObject *module, int fd)
     int ret;
     char name[MAXPATHLEN+1];
 
-    ret = ptsname_r(fd, name, sizeof(name));
+    if (HAVE_PTSNAME_R_RUNTIME) {
+        ret = ptsname_r(fd, name, sizeof(name));
+    }
+    else {
+        ret = -1;
```

Comment:
Can you use ptsname() if ptsname_r() is not available?

Reply:
What is errno=-1? Can it be replaced with a constant?

Reply:
Yes, I was also thinking about rewritting this. We need to fallback to `ptsname()` when `ptsname_r()` is not available in runtime.

Reply:
Done!